src/delay.rs

```diff
@@ -47,17 +47,19 @@ impl DelayMs<u8> for Delay {
 
 impl DelayUs<u32> for Delay {
     fn delay_us(&mut self, us: u32) {
-        let rvr = us * (self.clocks.sysclk().0 / 1_000_000);
+        let mut total_rvr = us * (self.clocks.sysclk().0 / 1_000_000);
```

Comment:
Depending on the system clock this might not do what one expects as it can overflow, I'd recommend to extend to `u64` and to use round to nearest (for completeness) as follows.

```suggestion
        let mut total_rvr = ((us as u64 * self.clocks.sysclk().0 as u64 + 500_000) / 1_000_000) as u32 ;
```